File: frontend/src/types/portfolio.ts
export interface EquityPoint {
  date: string;
  equity: number;
  pnl: number;
}

export interface StreakInfo {
  current_win_streak: number;
  current_loss_streak: number;
  best_win_streak: number;
  worst_loss_streak: number;
}

export interface PortfolioStats {
  user_id: string;
  initial_capital: number;
  current_equity: number;
  total_pnl: number;
  total_pnl_pct: number;
  total_trades: number;
  winning_trades: number;
  losing_trades: number;
  win_rate: number;
  avg_win: number;
  avg_loss: number;
  profit_factor: number;
  sharpe_ratio: number;
  max_drawdown_pct: number;
  best_trade: number;
  worst_trade: number;
  streaks: StreakInfo;
  equity_curve: EquityPoint[];
}

File: backend/app/schemas/portfolio.py
from pydantic import BaseModel


class EquityPoint(BaseModel):
    date: str
    equity: float
    pnl: float


class StreakInfo(BaseModel):
    current_win_streak: int
    current_loss_streak: int
    best_win_streak: int
    worst_loss_streak: int


class PortfolioStats(BaseModel):
    user_id: str
    initial_capital: float
    current_equity: float
    total_pnl: float
    total_pnl_pct: float
    total_trades: int
    winning_trades: int
    losing_trades: int
    win_rate: float
    avg_win: float
    avg_loss: float
    profit_factor: float
    sharpe_ratio: float
    max_drawdown_pct: float
    best_trade: float
    worst_trade: float
    streaks: StreakInfo
    equity_curve: list[EquityPoint]
